docs/building-images.md

```diff
@@ -57,6 +57,34 @@ operatingSystem:
     to generate the value for this field).
   * `sshKey` - Optional; Full public SSH key to configure for the user.
 
+### Elemental
+
+The elemental configuration section is entirely optional.
+
+The following describes the options for the elemental section:
+```yaml
+elemental:
+  registration:
+    url: https://rancher/elemental/registration/8hwjpp8tj56n7fh6x57ns7cd2tv6vxnbzg79zwmqzxdql2lm92xslr
+    ca-cert: |-
+      -----BEGIN CERTIFICATE-----
+      MIIBvDCCAWOgAwIBAgIBADAKBggqhkjOPQQDAjBGMRwwGgYDVQQKExNkeW5hbWlj
+      -----END CERTIFICATE-----
+    auth: tpm
+    emulate-tpm: true
+    emulated-tpm-seed: 1
+```
+
+* `url` - Required; Endpoint URL for registration.
+* `ca-cert` - Required; Certificate for TLS verification.
+
+For information on the TPM configuration, see the [Elemental MachineRegistration documentation](https://elemental.docs.rancher.com/machineregistration-reference/#configelementalregistration).
+
+Additionally, the following RPMs must be included in the RPMs directory as described in the
+Image Configuration Directory section:
+* `elemental-register`
+* `elemental-system-agent`
```

Comment:
I guess this is fine for the initial implementation but I definitely don't feel it makes it a great user experience. Probably pre-installing and detecting image arch (similarly to nmc) would be a much preferred approach but I'd leave this for PM to decide.

Reply:
I agree, I don't love it either. My concern about embedding it into the EIB container image is that it locks the user into a specific version. To work around that, we'd need to provide a way to override which would look similar to this, so it was a judgement call based on lack of time.